### PR TITLE
Use DOI.org content negotiation only, fallback to URL metadata for DOIs

### PR DESCRIPTION
**Pull request by dhimmel**
refs https://github.com/manubot/manubot/issues/309
refs https://github.com/crosscite/content-negotiation/issues/104

DataCite has stopped returning content negotiation results for Crossref DOIs. This change might be temporary or permanent, awaiting further deliberation.

### CODE REVIEW

File: manubot/cite/doi.py [hunk 1]
```diff
@@ -146,11 +146,21 @@ def get_doi_csl_item_negotiation(
 
 
 def get_doi_csl_item_datacite(doi: str):
-    return get_doi_csl_item_negotiation(doi, content_negotiation_url_datacite)
+    """
+    As of 2021-01, the DataCite Content Negotiation restricted
```
Comment (agitter) [inline manubot/cite/doi.py:150]:
```suggestion
    As of 2022-01, the DataCite Content Negotiation restricted
```

File: manubot/cite/doi.py [hunk 2]
```diff
@@ -210,7 +231,7 @@ def get_doi_csl_item(doi: str):
 
 
 doi_retrievers = [
```
Comment (agitter) [inline manubot/cite/doi.py:233]: We could reiterate with a comment here why `get_doi_csl_item_datacite` is not in the list.

File: manubot/cite/doi.py [hunk 3]
```diff
@@ -210,7 +231,7 @@ def get_doi_csl_item(doi: str):
...
-    get_doi_csl_item_datacite,
-    get_doi_csl_item_crosscite,
+    get_doi_csl_item_default,
+    get_doi_csl_item_url,
```
Comment (agitter) [inline manubot/cite/doi.py:235]: Do you expect the URL-based retrieval to be more accurate than Zotero's DOI support?

**Review by agitter — CHANGES_REQUESTED** [review]
That's an inconvenient change.  I expect this could disrupt a lot of existing Manubot manuscripts.

The `test_cite_filter.py` test failure looks like it may be an actual failure due to changed behavior.

File: manubot/pandoc/tests/test_cite_filter/output.txt [hunk 4]
```diff
@@ -10,52 +10,42 @@ Defining citekeys with the link reference syntax [3,4,5].
...
 1. Orthodontic treatment with tooth transplantation for patients with cleft lip and palate.
-Kotaro Tanimoto, Tamami Yanagida, Kazuo Tanne
-The Cleft palate-craniofacial journal : official publication of the American Cleft Palate-Craniofacial Association (2010-09) https://www.ncbi.nlm.nih.gov/pubmed/20170387
+Kotaro Tanimoto, Tamami Yanagida, Kazuo Tanne The Cleft palate-craniofacial journal : official publication of the American Cleft Palate-Craniofacial Association (2010-09) https://www.ncbi.nlm.nih.gov/pubmed/20170387
 DOI: 10.1597/08-134 · PMID: 20170387
 
 2. Lung cancer incidence decreases with elevation: evidence for oxygen as an inhaled carcinogen
-Kamen P. Simeonov, Daniel S. Himmelstein
-PeerJ (2015-01-13) https://doi.org/98p
+Kamen P Simeonov, Daniel S Himmelstein PeerJ (2015-01-13) https://doi.org/98p
```
Comment (dhimmel) [inline manubot/pandoc/tests/test_cite_filter/output.txt:24]: Will have to look into the missing newline between author and venue in md output. Related to https://github.com/jgm/pandoc/issues/6921

File: manubot/cite/doi.py (hunk 2 above)
Comment (dhimmel) [reply manubot/cite/doi.py:233]: done in https://github.com/manubot/manubot/pull/319/commits/813fcd6dfc7b87f836ebf29b89e2705b1596b5bd

**Comment by dhimmel** [comment]
Ah the following update was just [announced](https://status.datacite.org/incidents/k7rjjq59kpyb) by DataCite:

> Requests for DataCite DOIs should be unaffected going forward, requests for Crossref DOIs against DataCite Content negotiation directly may occasionally return 404s when it is unable to process.
> It is preferred if possible to use Content Negotiation via doi.org and you will be redirected as appropriate to the registration agencies content negotiation service as appropriate.

But when I query DataCite CN URLs for Crossref DOIs, they're still failing.

File: manubot/cite/doi.py [hunk 5]
```diff
@@ -210,7 +231,19 @@ def get_doi_csl_item(doi: str):
...
+On retreiver ordering
+
+1. get_doi_csl_item_default: try DOI metadata first as per the Registration Agency's Content Negotiation.
+2. get_doi_csl_item_url: scrape data from the website where the DOI resolves.
+3. get_doi_csl_item_zotero: use Zotero translation-server to get DOI metadata.
+   Placed last since it's unlikely to work when get_doi_csl_item_default fails,
+   since it also uses content negotiation.
+
+get_doi_csl_item_datacite is not included because as of 2022-01 it only works for DataCite DOIs,
+and get_doi_csl_item_default ends up redirrecting to this endpoint for DataCite registered DOIs.
```
Comment (agitter) [inline manubot/cite/doi.py:248]:
```suggestion
and get_doi_csl_item_default ends up redirecting to this endpoint for DataCite registered DOIs.
```

**Review by agitter — APPROVED** [review]
We're seeing the
```
## WARNING
OpenTimestamp command returned nonzero code (-6).
>>> ots stamp webpage/v/testing/index.html
```
failure on macOS again.  It's unrelated to these changes, so I'm approving.

Note one last spelling change before merging.

Edit: reported in https://github.com/manubot/manubot/issues/320

**Pull request merged**